nbgrader/install.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import print_function, absolute_import
+
+"""
+Misc utils to deals wit installing nbgrader on a system
+"""
+
+import sys
+import io
+import os.path
+import json
+import argparse
+
+import IPython.html.nbextensions as nbe
+from IPython.utils.path import locate_profile, get_ipython_dir
+from IPython.core.profiledir import ProfileDir
+from IPython.utils.py3compat import cast_unicode_py2
+
+
+def install(profile='default', symlink=True, user=False, prefix=None,
+            verbose=False, path=None, activate=False, deactivate=False):
+    """Install and activate nbgrader on a profile
+    """
+
+    dname = os.path.dirname(__file__)
+    # might want to check if already installed and overwrite if exist
+    if symlink and verbose:
+        print('I will try to symlink the extension instead of copying files')
+    if prefix and verbose:
+        print("I will install in prefix:", prefix)
+    if not path:
+        nbextensions_dir = None
+    else:
+        nbextensions_dir = os.path.join(path,'nbextensions')
+    nbe.install_nbextension(os.path.join(dname, 'nbextensions', 'nbgrader'),
+                            user=user,
+                            prefix=prefix,
+                            symlink=symlink,
+                            nbextensions_dir=nbextensions_dir
+                            )
+    _activate(profile, ipythondir=path)
+
+
+# TODO pass prefix as argument.
+def _activate(profile, ipythondir=None):
+    """
+    Manualy modify the frontend json-config to load nbgrader extension.
+    """
+
+    if not ipythondir:
+        pdir = ProfileDir.find_profile_dir_by_name(get_ipython_dir(), profile).location
+    else :
+        ipythondir = os.path.expanduser(ipythondir)
+        pdir = ProfileDir.find_profile_dir_by_name(ipythondir, profile).location
+
+    json_dir = os.path.join(pdir, 'nbconfig')
+    json_file = os.path.join(pdir, 'nbconfig', 'notebook.json')
+    json_file = os.path.expanduser(json_file)
+
+    try:
+        with io.open(json_file, 'r') as f:
+            config = json.loads(f.read())
+    except IOError:
+        # file just don't exists yet. IPython might have never been launched.
+        config = {}
+
+    if not config.get('load_extensions', None):
+        config['load_extensions'] = {}
+    config['load_extensions']['nbgrader/nbgrader'] = True
```

Comment:
It seems weird to me to have to activate it with `nbgrader/nbgrader` rather than just `nbgrader`. Perhaps something clearer would be to rename the extension itself to something like `create_assignment` (as technically that is what it's called), so then when you load it you would be loading `nbgrader/create_assignment`.